tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQIntegration.cs

```diff
@@ -88,22 +89,23 @@ internal static string GetOperationName(Tracer tracer, string spanKind)
             };
         }
 
-        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IDictionary<string, object> headers)
+        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IBody body, IBasicProperties basicProperties)
         {
             var dataStreamsManager = tracer.TracerManager.DataStreamsManager;
-            if (dataStreamsManager == null || !dataStreamsManager.IsEnabled || headers == null)
+            if (dataStreamsManager == null || basicProperties == null || !dataStreamsManager.IsEnabled || basicProperties.Headers == null)
             {
                 return;
             }
 
             try
             {
-                var headersAdapter = new RabbitMQHeadersCollectionAdapter(headers);
+                var headersAdapter = new RabbitMQHeadersCollectionAdapter(basicProperties.Headers);
                 var edgeTags = string.IsNullOrEmpty(tags.Exchange) ?
                                    // exchange can be empty for "direct"
                                    new[] { "direction:out", $"topic:{tags.Queue ?? tags.RoutingKey}", "type:rabbitmq" } :
                                    new[] { "direction:out", $"exchange:{tags.Exchange}", $"has_routing_key:{!string.IsNullOrEmpty(tags.RoutingKey)}", "type:rabbitmq" };
-                span.SetDataStreamsCheckpoint(dataStreamsManager, CheckpointKind.Produce, edgeTags);
+                // size of headers is ignored for now
+                span.SetDataStreamsCheckpoint(dataStreamsManager, CheckpointKind.Produce, edgeTags, body?.Length ?? 0, 0);
```

Comment:
Note that `body` can't be null as it's duck typed, so you need to check the underlying object